explorer/src/components/account/history/TokenTransfersDetails.tsx

```diff
@@ -0,0 +1,213 @@
+    if (parsed.meta?.preTokenBalances) {
+      parsed.meta.preTokenBalances.forEach((balance) => {
+        const account =
+          parsed.transaction.message.accountKeys[balance.accountIndex];
+        mintMap.set(account.pubkey.toBase58(), {
+          decimals: balance.uiTokenAmount.decimals,
+          mint: balance.mint,
+        });
+      });
+    }
+
+    if (parsed.meta?.postTokenBalances) {
+      parsed.meta.postTokenBalances.forEach((balance) => {
+        const account =
+          parsed.transaction.message.accountKeys[balance.accountIndex];
+        mintMap.set(account.pubkey.toBase58(), {
+          decimals: balance.uiTokenAmount.decimals,
+          mint: balance.mint,
+        });
+      });
+    }
+
+    // Extract all transfers from transaction
+    let transfers: (Transfer | TransferChecked)[] = [];
+    InstructionContainer.create(parsed).instructions.forEach(
+      ({ instruction, inner }) => {
+        const transfer = getTransfer(instruction);
+        if (transfer) {
+          transfers.push(transfer);
+        }
+        inner.forEach((instruction) => {
+          const transfer = getTransfer(instruction);
+          if (transfer) {
+            transfers.push(transfer);
+          }
+        });
+      }
+    );
+
+    // Filter out transfers not belonging to this mint
+    transfers = transfers.filter((transfer) => {
+      if ("tokenAmount" in transfer && transfer.mint !== pubkey) {
+        return false;
+      } else if (
+        mintMap.has(transfer.source.toBase58()) &&
+        mintMap.get(transfer.source.toBase58())?.mint !== pubkey.toBase58()
+      ) {
+        return false;
+      } else if (
+        mintMap.has(transfer.destination.toBase58()) &&
+        mintMap.get(transfer.destination.toBase58())?.mint !== pubkey.toBase58()
+      ) {
+        return false;
+      }
+      return true;
+    });
+
+    transfers.forEach((transfer) => {
+      let units = "Tokens";
+      let amountString = "";
+
+      if (mintDetails?.symbol) {
+        units = mintDetails.symbol;
+      }
+
+      if ("tokenAmount" in transfer) {
+        amountString = transfer.tokenAmount.uiAmountString;
+      } else {
+        let decimals = 0;
+
+        if (mintDetails?.decimals) {
+          decimals = mintDetails.decimals;
+        } else if (mintMap.has(transfer.source.toBase58())) {
+          decimals = mintMap.get(transfer.source.toBase58())?.decimals || 0;
+        } else if (mintMap.has(transfer.destination.toBase58())) {
+          decimals =
+            mintMap.get(transfer.destination.toBase58())?.decimals || 0;
+        }
+
+        amountString = new Intl.NumberFormat("en-US", {
+          minimumFractionDigits: decimals,
+          maximumFractionDigits: decimals,
+        }).format(normalizeTokenAmount(transfer.amount, decimals));
+      }
+
+      detailsList.push(
+        <tr key={signature + transfer.source + transfer.destination}>
```

Comment:
One transaction could have multiple transfers between the same accounts. Maybe each transfer could have a unique id from signature + instruction / inner indexes?

Reply:
That makes sense! I'll add that.